src/EventSourcingDb/DependencyInjection/ServiceCollectionExtensions.cs

```diff
@@ -0,0 +1,36 @@
+﻿using System;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+
+namespace EventSourcingDb.DependencyInjection;
+
+public static class ServiceCollectionExtensions
+{
+    public static void AddEventSourcingDb(
+        this IServiceCollection services,
+        IConfiguration configuration,
+        Action<EventSourcingDbOptions>? configureOptions = null
+    )
+    {
+        services
+            .AddOptions<EventSourcingDbOptions>()
+            .Bind(configuration.GetSection("EventSourcingDb"))
+            .ValidateOnStart();
+
+        if (configureOptions is not null)
+        {
+            services.PostConfigure(configureOptions);
+        }
+
+        services.AddScoped<Client>(sp =>
```

Comment:
The big question always is: Create an interface (and bind against it) or not? I personally want to have interfaces from third-party libs so that I can setup proper mocks in tests. But there are other opinions out there.